File: 41 - Find the Majority Element/solution.js
function majorityElement(nums) {
  // Write your code here
  let count = 0;
  let majority = null;

  for (let num of nums) {
    if (count === 0) {
      majority = num;
    }
    count += num === majority ? 1 : -1;
  }

  let freq = nums.filter((num) => num === majority).length;
  return freq > Math.floor(nums.length / 2) ? majority : null;
}

// Test Cases
console.log(majorityElement([3, 3, 4, 2, 4, 4, 2, 4, 4]));

File: 41 - Find the Majority Element/solution.py
def majority_element(nums):
    # Write your code here
    count = 0
    majority = None

    for num in nums:
        if count == 0:
            majority = num
        count =  count + 1 if num == majority else count - 1

    freq = sum(1 for num in nums if num == majority)
    return freq > len(nums) // 2 and majority or None

# Test Cases
print(majority_element([3, 3, 4, 2, 4, 4, 2, 4, 4]))